File: example/typescript/OtherModule/Fixtures/Asserts.ts
namespace OtherModule.Assert {
    export interface ExpectedOtherClass {
        id?: string,
        amount?: number,
    }
    export function otherClass(given: OtherClass, expected: ExpectedOtherClass) {
        if (expected.id !== undefined) {
            AssertEquals(given.id.value, expected.id)
        }
        if (expected.amount !== undefined) {
            AssertEquals(given.amount, expected.amount)
        }
    }
}

File: example/kotlin/src/testFixtures/kotlin/com/some/pkg/othermodule/Asserts.kt
package com.some.pkg.othermodule

import com.some.pkg.othermodule.api.OtherClass
import org.assertj.core.api.Assertions.assertThat

data class ExpectedOtherClass(
    var id: String? = null,
    var amount: Int? = null,
)
fun assertOtherClass(given: OtherClass, expectedInit: ExpectedOtherClass.() -> Unit) {
    val expected = ExpectedOtherClass().apply(expectedInit)

    expected.id?.let {
        assertThat(given.id.value).isEqualTo(it)
    }

    expected.amount?.let {
        assertThat(given.amount).isEqualTo(it)
    }
}